### PR TITLE
For #251: Listing plugins with filter

### PR DESCRIPTION
**Pull request by bkuzmic**
For #251: 

- listing Plugins with filters
- created all other support interfaces/classes
- created a new puzzle to finish Plugin implementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#264](https://github.com/amihaiemil/docker-java-api/issues/264) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#264](https://github.com/amihaiemil/docker-java-api/issues/264) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/b3e34d2c-7bf4-4b16-9aeb-b1a6018fe5d3); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/amihaiemil/docker/ListedPluginsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                ),
+                new Condition(
+                    "iterate() must send a GET request",
+                    req -> "GET".equals(req.getRequestLine().getMethod())
+                ),
+                new Condition(
+                    //@checkstyle LineLength (1 line)
+                    "iterate() query parameters must include the filters provided",
+                    req -> {
+                        // @checkstyle LineLength (1 line)
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/ListedPluginsTestCase.java]: @bkuzmic Let's rearrange this so we do not need to use this second checkstyle ignore tag

File: src/test/java/com/amihaiemil/docker/RtPluginTestCase.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    HttpStatus.SC_OK,
+                    Json.createObjectBuilder()
+                        .add("Name", "p1")
+                        .add("Driver", "custom")
+                        .add("Enabled", true)
+                        .build().toString()
+                ),
+                new Condition(
+                    "Method should be a GET",
+                    req -> req.getRequestLine().getMethod().equals("GET")
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtPluginTestCase.java]: @bkuzmic For uniformity's sake, lets invert this comparison like you've made before: `"GET".equals(...)`

File: src/main/java/com/amihaiemil/docker/RtPlugin.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    /**
+     * Apache HttpClient which sends the requests.
+     */
+    private final HttpClient client;
+
+    /**
+     * Base URI.
+     */
+    private final URI baseUri;
```
Comment (paulodamaso) [inline src/main/java/com/amihaiemil/docker/RtPlugin.java]: @bkuzmic I'd use just `uri` here

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Rate for @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) is not set

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/amihaiemil/docker-java-api/issues/264#issuecomment-451550953))

@paulodamaso This is what I know about this job in G6LPQQV2P, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#264](https://github.com/amihaiemil/docker-java-api/issues/264) is [in scope](http://www.zerocracy.com/policy.html#14) for 8hr
  * The role is `REV`
  * The job is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) for [8hr](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @bkuzmic[/z](https://www.0crat.com/u/bkuzmic): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/G6LPQQV2P?q=%7Bjob%3A%27gh%3Aamihaiemil%2Fdocker-java-api%23264%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by paulodamaso** [comment]
@amihaiemil please take a look: the job was closed but @0crat did not paid me and it's still in my agenda

**Comment by 0crat** [comment]
Are you speaking to me or about me [here](https://github.com/amihaiemil/docker-java-api/issues/264#issuecomment-451551480); you must always start your message with my name if you want to address it to me, see [§1](http://www.zerocracy.com/policy.html#1)

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**